lib/results.js

```diff
@@ -44,6 +45,21 @@ function warning(message) {
     };
 }
 
+function error(exception) {
+    if (typeof exception == 'string') {
```

Comment:
I don't think this branch is needed -- could we just assume that an error will always be passed in?